### PR TITLE
updated account delete

### LINKED ISSUE #479: Anonymise the user data when a user is deleted PART 2
anonymize the identity of the user to be truly deleted and anonym

transform the email of the user in something like: 
anonymous1692102521871@dotkernel.com

### PR DESCRIPTION
**Pull request by bidi47**

### CODE REVIEW

**Comment by bidi47** [comment]
other changes
- moved some messages to the Message class (authentication)
- added the messages to the homepage - in this case, to confirm the account deletion

File: src/User/src/Service/UserService.php [hunk 1]
```diff
@@ -134,6 +135,16 @@ public function updateUser(User $user, array $data = []): User
 
         if (isset($data['isDeleted'])) {
             $user->setIsDeleted((bool) $data['isDeleted']);
+
+            if ((bool) $data['isDeleted'] === true) {
+                // make user anonymous
+                $user->setIdentity('anonymous' . date('dmYHis') . $this->config['userAnonymizeAppend']);
```
Comment (alexmerlin) [inline src/User/src/Service/UserService.php]:
```suggestion
                $user->setIdentity(
                    sprintf('anonymous%s@%s', date('dmYHis'), $this->config['userAnonymizeAppend'])
                );
```
You had a missing `@` between the date and the configured append part.
Also, for better readability, I suggest using `sprintf`.

File: config/autoload/local.php.dist [hunk 2]
```diff
@@ -55,16 +55,17 @@ return [
...
         'verifyUrl'      => 'https://www.google.com/recaptcha/api/siteverify',
     ],
-    'rememberMe'  => [
+    'rememberMe'          => [
         'cookie' => [
             'name'     => 'rememberMe',
             'lifetime' => 3600 * 24 * 30,
         ],
     ],
+    'userAnonymizeAppend' => '',
```
Comment (alexmerlin) [inline config/autoload/local.php.dist:70]: FYI, I created an issue in the frontend documentation repo, to document the user anonymization and this new config value.
https://github.com/dotkernel/frontend-documentation/issues/6

(not a blocker for this issue, just a thing for us to keep in mind)

**Review by alexmerlin — CHANGES_REQUESTED** [review]

File: src/User/src/Service/UserService.php (hunk 1 above)
Comment (bidi47) [reply src/User/src/Service/UserService.php]: i just assumed the config (userAnonymizeAppend) would have the '@'
either way, your solution is better

File: config/autoload/local.php.dist (hunk 2 above)
Comment (bidi47) [reply config/autoload/local.php.dist:70]: got it

File: src/User/src/Service/UserService.php (hunk 1 above)
Comment (alexmerlin) [reply src/User/src/Service/UserService.php]: ~~When I tested locally, I forgot that @arhimede's [comment](https://github.com/dotkernel/frontend/issues/479#issuecomment-2331111238) included both the `@` symbol and the domain name.
I just set `userAnonymizeAppend` to `dotkernel.com` and after testing, I ended up with an anonymized email without the `@` in it.~~

@arhimede's [comment](https://github.com/dotkernel/frontend/issues/479#issuecomment-2331111238) said:

> Move the string "dotkernel.com" in config, in local.php

That's why I did not include the `@` symbol in my config.

Your solution (concatenating strings) was just as correct as mine, but IMHO the `sprintf` solution looks more readable than concatenating four strings.

**Review by alexmerlin — APPROVED** [review]

**Pull request merged**